app/src/androidTest/java/com/cmput301w23t09/qrhunter/profile/TestProfileFragment.java

```diff
@@ -0,0 +1,55 @@
+package com.cmput301w23t09.qrhunter.profile;
+
+import static org.junit.Assert.assertTrue;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.rule.ActivityTestRule;
+import com.cmput301w23t09.qrhunter.GameActivity;
+import com.cmput301w23t09.qrhunter.GameController;
+import com.cmput301w23t09.qrhunter.R;
+import com.robotium.solo.Solo;
+import junit.framework.TestCase;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+/** Test classes for profile activity */
+public class TestProfileFragment {
+  private Solo solo;
+
+  @Rule
+  public ActivityTestRule<GameActivity> rule = new ActivityTestRule(GameActivity.class, true, true);
+
+  /**
+   * Runs before all tests and creates solo instance
+   *
+   * @throws Exception
+   */
+  @Before
+  public void setUp() throws Exception {
+    // create solo
+    solo = new Solo(InstrumentationRegistry.getInstrumentation(), rule.getActivity());
+    // navigate to profile fragment
+    solo.clickOnView(solo.getView(R.id.navigation_my_profile));
```

Comment:
In setup, try this:
```
activityScenarioRule
        .getScenario()
        .onActivity(
            activity -> {
              activity.sendBroadcast(new Intent(Intent.ACTION_CLOSE_SYSTEM_DIALOGS));
              solo = new Solo(InstrumentationRegistry.getInstrumentation(), activity);
            });
```
and change `public ActivityTestRule<GameActivity> rule = new ActivityTestRule(GameActivity.class, true, true);
` to `public ActivityScenarioRule<GameActivity> activityScenarioRule = new ActivityScenarioRule<>(GameActivity.class);`